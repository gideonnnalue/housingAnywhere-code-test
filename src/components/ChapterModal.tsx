import { FC } from "react";
import { ICharacterModalProps, EpisodeData } from "./types";

const ChapterModal: FC<ICharacterModalProps> = ({
  episodes,
  closeModal,
}): JSX.Element => {
  return (
    <div className="modal__backdrop">
      <div className="modal__body" data-testid="modal-body">
        <button className="modal__close" onClick={closeModal}>
          X
        </button>
        <h3>CHAPTERS CHARACTER FEATURED IN</h3>
        <div className="characterCard__info--text" data-testid="chapter-list">
          {episodes.map((episode: EpisodeData) => (
            <p key={episode.id} data-testid="chapter-item">
              {episode.episode} - <span>{episode.name}</span>,
            </p>
          ))}
        </div>
      </div>
    </div>
  );
};

export default ChapterModal;
