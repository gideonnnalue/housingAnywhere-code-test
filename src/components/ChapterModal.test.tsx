import { render, screen, cleanup } from "@testing-library/react";
import ChapterModal from "./ChapterModal";

const episodes = [
  {
    air_date: "January 27, 2014",
    characters: [
      "https://rickandmortyapi.com/api/character/1",
      "https://rickandmortyapi.com/api/character/2",
    ],
    created: "2017-11-10T12:56:34.339Z",
    episode: "S01E06",
    id: 6,
    name: "Rick Potion #9",
    url: "https://rickandmortyapi.com/api/episode/6",
  },
  {
    air_date: "January 27, 2014",
    characters: [
      "https://rickandmortyapi.com/api/character/1",
      "https://rickandmortyapi.com/api/character/2",
    ],
    created: "2017-11-10T12:56:34.339Z",
    episode: "S01E06",
    id: 7,
    name: "Rick Potion #9",
    url: "https://rickandmortyapi.com/api/episode/6",
  },
];

beforeEach(() => {
  render(<ChapterModal episodes={episodes} closeModal={() => {}} />);
});

afterEach(cleanup);

describe("Chapter Modal Test", () => {
  test("Should take snapshot of modal", () => {
    const { asFragment } = render(
      <ChapterModal episodes={episodes} closeModal={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("Should have two elements", () => {
    const modalBody = screen.getByTestId("modal-body");
    const chapterList = screen.getByTestId("chapter-list");
    const chapterItems = screen.getAllByTestId("chapter-item");

    expect(modalBody).toContainElement(chapterList);
    expect(chapterItems).toHaveLength(2);
    expect(chapterItems[0]).toHaveTextContent("S01E06");
    expect(chapterItems[0]).toContainHTML("<span>Rick Potion #9</span>");
  });
});
