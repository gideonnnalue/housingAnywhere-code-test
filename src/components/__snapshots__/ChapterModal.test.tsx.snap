// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chapter Modal Test Should take snapshot of modal 1`] = `
<DocumentFragment>
  <div
    class="modal__backdrop"
  >
    <div
      class="modal__body"
      data-testid="modal-body"
    >
      <button
        class="modal__close"
      >
        X
      </button>
      <h3>
        CHAPTERS CHARACTER FEATURED IN
      </h3>
      <div
        class="characterCard__info--text"
        data-testid="chapter-list"
      >
        <p
          data-testid="chapter-item"
        >
          S01E06 - 
          <span>
            Rick Potion #9
          </span>
          ,
        </p>
        <p
          data-testid="chapter-item"
        >
          S01E06 - 
          <span>
            Rick Potion #9
          </span>
          ,
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
